lib/sles4sap_publiccloud.pm

```diff
@@ -608,4 +610,63 @@ sub delete_network_peering {
     }
 }
 
+=head2 get_hana_database_info
+
+    Parses "hdbsql" command output.
+    Returns 1 if hana database is online otherwise 0 offline.
+
+=over 1
+
+=item B<RESULT> - default "-1" representing empty value.
+
+=back
+=cut
+
+sub get_hana_database_info {
+    my ($self) = @_;
+    my $result = -1;
+    $self->is_hana_online();
+    my $output_cmd = $self->run_cmd(cmd => "hdbsql -u SYSTEM -p Linux1234 -i 00 'SELECT * FROM SYS.M_DATABASES;'", runas => get_required_var("SAP_SIDADM"));
+    my $hana_database_online = '"HDB","HDB-00","YES","ACTIVE","","","DEFAULT","?"';
+
+    $result = ($output_cmd =~ /$hana_database_online/ ? 1 : 0);
+    record_info("Hana database status", $self->run_cmd(cmd => "hdbsql -u SYSTEM -p Linux1234 -i 00 'SELECT * FROM SYS.M_DATABASES;'"));
+    return $result;
+}
+
+=head2 is_hana_database_offline
+
+    Check if hana DB is offline.
+    Returns if hana database is online with 1 and offline with 0
+
+=over 1
+
+=item B<TIMEOUT> - default 900
+
+=back
+=cut
+
+sub is_hana_database_offline {
+    my ($self, %args) = @_;
+    my $timeout = bmwqemu::scale_timeout($args{timeout} // 900);
+    my $start_time = time;
+    my $db_status = -1;
+    my $consecutive_passes = 0;
+    my $total_consecutive_passes = ($args{total_consecutive_passes} // 5);
+
+    while ($consecutive_passes < $total_consecutive_passes) {
+        $db_status = $self->get_hana_database_info();
+        if (time - $start_time > $timeout) {
+            record_info("Hana database after timeout", $self->run_cmd(cmd => "hdbsql -u SYSTEM -p Linux1234 -i 00 'SELECT * FROM SYS.M_DATABASES;'"));
+            die("Hana database still is online");
+        }
+        if ($db_status eq 0) {
+            last;
+        }
+        sleep 30;
```

Comment:
Is it OK that if checking for 5 consecutive passes and waiting for 30 seconds, the test could wait up to 2 minutes and 30 seconds before continuing? I suppose as written it's not such a big deal, as there's a exit clause for the loop at the first sight of an offline database, but wanted to point this out in case this was not the intention.

Reply:
It is intentional, thx for the comment

Reply:
Then please add a comment in the code why we need to wait so long. Code itself is not clear.